server/src/main/java/com/saralein/server/request/parser/RequestLineParser.java

```diff
@@ -0,0 +1,33 @@
+package com.saralein.server.request.parser;
+
+import com.saralein.server.request.transfer.RequestLine;
+
+import java.util.List;
+
+public class RequestLineParser {
+    public RequestLine parse(List<String> request) throws Exception {
```

Comment:
I feel like a "parse" method insinuates the thing it's being passed will be parsed. There's an extra layer of logic here in the method body where it has to know which item in the collection needs to get parsed.